packages/app/src/vis-packs/core/complex/MappedComplexLineVis.tsx

```diff
@@ -29,43 +32,68 @@ function MappedComplexLineVis(props: Props) {
   const {
     value,
     valueLabel,
+    dims,
+    dimMapping,
+    axisLabels,
+    axisValues,
+    title,
     toolbarContainer,
     config,
     lineConfig,
-    ...lineProps
   } = props;
 
-  const { dims } = lineProps;
   const { visType } = config;
+  const { customDomain, yScaleType, xScaleType, curveType, showGrid } =
+    lineConfig;
 
+  const [slicedDims, slicedMapping] = useSlicedDimsAndMapping(dims, dimMapping);
   const { phaseValues, amplitudeValues } = useMemo(
     () => getPhaseAmplitudeValues(value),
     [value],
   );
 
+  const [dataArray, dataForDomain] = useMappedArray(
+    visType === ComplexVisType.Amplitude ? amplitudeValues : phaseValues,
+    slicedDims,
+    slicedMapping,
+  );
+
+  const dataDomain = useDomain(dataForDomain, yScaleType) || DEFAULT_DOMAIN;
+  const visDomain = useVisDomain(customDomain, dataDomain);
+  const [safeDomain] = useSafeDomain(visDomain, dataDomain, yScaleType);
+
+  const xDimIndex = dimMapping.indexOf('x');
+  const ordinateLabel = valueLabel
+    ? `${valueLabel} (${COMPLEX_VIS_TYPE_LABELS[visType].toLowerCase()})`
+    : COMPLEX_VIS_TYPE_LABELS[visType];
+
   return (
     <>
       {toolbarContainer &&
         createPortal(
           <ComplexLineToolbar
-            disableAutoScale={dims.length <= 1} // with 1D datasets, `baseArray` and `dataArray` are the same so auto-scaling is implied
+            dataDomain={dataDomain}
             config={config}
             lineConfig={lineConfig}
           />,
           toolbarContainer,
         )}
 
-      <MappedLineVis
```

Comment:
Since I now need to pass a domain to the complex line toolbar here, I can't just let `MappedLineVis` compute it any more — I have to do it in this component. So I decided to remove the coupling between this component and `MappedLineVis`. It was the only "mapped" component doing something like this, so it's consistent with the others now.